packages/epo-react-lib/src/form/SelectListbox/hooks/useAccessibleDropdown.ts

```diff
@@ -0,0 +1,98 @@
+import { RefObject, useEffect, useRef, useState } from "react";
+import { onChangeCallback, Option } from "@/types/select-listbox";
+import { isSafari } from "../utilities";
+import { registerOpenDropdownHandlers } from "../handlers";
+
+interface useAccessibleDropdownArgs {
+  options: Option[];
+  value: string | string[] | null;
+  onChangeCallback: onChangeCallback;
+  isMultiselect: boolean;
+}
+
+interface useAccessibleDropdownResult {
+  isDropdownOpen: boolean;
+  setIsDropdownOpen: (v: boolean) => void;
+  activeIndex: number;
+  setActiveIndex: (v: number) => void;
+  select: (value: string) => void;
+  setIsFocus: (v: boolean) => void;
+  listRef: RefObject<HTMLUListElement>;
+  buttonRef: RefObject<HTMLButtonElement>;
+}
+
+const useAccessibleDropdown: (
+  props: useAccessibleDropdownArgs
+) => useAccessibleDropdownResult = ({
+  options,
+  value,
+  onChangeCallback,
+  isMultiselect,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpenInternal] = useState(false);
+  const listRef = useRef<HTMLUListElement>(null);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [isFocus, setIsFocus] = useState(false);
+
+  const select = (selectedValue: string | null) => {
+    if (onChangeCallback) {
+      if (isMultiselect && selectedValue) {
```

Comment:
these lines do a lot of work in adding the new value OR removing it if it's selected && multiselect OR if it's multiselect but the current value is a string.